packages/collection-model/lib/extended-model.js

```diff
@@ -1,89 +1,108 @@
-var MongoDBCollection = require('./collection-model');
-var toNS = require('mongodb-ns');
-var raf = require('raf');
+const AmpersandModel = require('ampersand-model');
+const AmpersandCollection = require('ampersand-collection');
+const { promisify } = require('util');
+const toNs = require('mongodb-ns');
 
-/**
- * Metadata for a MongoDB Collection.
- * @see http://npm.im/mongodb-collection-model
- */
-var CollectionModel = MongoDBCollection.extend({
-  namespace: 'MongoDBCollection',
-  session: {
-    selected: {
-      type: 'boolean',
-      default: false
-    }
-  },
-  derived: {
-    name: {
-      deps: ['_id'],
-      fn: function() {
-        return toNS(this._id).collection;
-      }
-    },
-    specialish: {
-      deps: ['_id'],
-      fn: function() {
-        return toNS(this._id).specialish;
-      }
-    },
-    url: {
-      deps: ['_id'],
-      fn: function() {
-        return `/collections/${this.getId()}`;
-      }
-    }
+const CollectionModel = AmpersandModel.extend({
+  modelType: 'Collection',
+  idAttribute: '_id',
+  props: {
+    _id: 'string',
+
+    // Normalized values from listCollections command
+    name: { type: 'string', required: true },
+    database: { type: 'string', required: true },
+    type: { type: 'string', required: true },
+    system: { type: 'boolean', required: true },
+    oplog: { type: 'boolean', required: true },
+    command: { type: 'boolean', required: true },
+    special: { type: 'boolean', required: true },
+    specialish: { type: 'boolean', required: true },
+    normal: { type: 'boolean', required: true },
+    readonly: 'boolean',
+    collation: 'object',
+    pipeline: 'array',
+    validation: 'object',
+
+    // Normalized values from collStats command
+    ns: 'string',
+    is_capped: 'boolean',
+    max: 'number',
+    is_power_of_two: 'boolean',
+    index_sizes: 'object',
+    document_count: 'number',
+    document_size: 'number',
+    storage_size: 'number',
+    index_count: 'number',
+    index_size: 'number',
+    padding_factor: 'number',
+    extent_count: 'number',
+    extent_last_size: 'number',
+    flags_user: 'number',
+    max_document_size: 'number',
+    size: 'number',
+    index_details: 'object',
+    wired_tiger: 'object',
   },
-  serialize: function() {
-    return this.getAttributes(
-      {
-        props: true
-      },
-      true
+  /**
+   * @param {{ dataService: import('mongodb-data-service').DataService }} dataService
+   * @returns
+   */
+  async fetch({ dataService }) {
+    const collectionStatsAsync = promisify(
+      dataService.collectionStats.bind(dataService)
     );
+    const collectionInfoAsync = promisify(
+      dataService.collectionInfo.bind(dataService)
+    );
+    const [collStats, collectionInfo] = await Promise.all([
+      collectionStatsAsync(this.getId()),
```

Comment:
was this moved from somewhere else?

Reply:
Sorry, not sure what exactly you're referring to 🙇 The whole method or this particular line?